src/prezzemolo/graph.py

```diff
@@ -96,3 +155,13 @@ def find_shortest_path(self, start: Vertex[ValueType], end: Vertex[ValueType], r
                 return reversed(result)
             return iter(result)
         return None
+
+    def find_weighted_shortest_path(self, start: Vertex[ValueType], end: Vertex[ValueType], reverse: bool = True) -> Optional[Iterator[Vertex[ValueType]]]:
+        # start and end are type checked inside _dijkstra_search()
+        vertex_2_parent: Dict[Vertex[ValueType], Optional[Vertex[ValueType]]] = {vertex: None for vertex in self.__vertexes}
```

Comment:
You can just leave the dict empty here.

Reply:
How about just shortening it to:
```
vertex_2_parent: Dict[Vertex[ValueType], Optional[Vertex[ValueType]]] = {start: None}
```
I think only the start vertex needs initialization. This line would save two lines in `_dijkstra()`